shoop/testing/factories.py

```diff
@@ -548,8 +557,12 @@ def create_random_order(customer=None, products=(), completion_probability=0):
     if not customer:
         raise ValueError("No valid contacts")
 
-    request = RequestFactory().get("/")
-    request.customer = customer
+    shop = get_default_shop()
+
+    request = apply_request_middleware(RequestFactory().get("/"),
+                                       shop=get_default_shop(),
```

Comment:
You shouldn't need to pass `shop` here -- the ShopMiddleware should always use the default shop...